components/payment.js

```diff
@@ -132,6 +150,8 @@ export const useWebLnPayment = () => {
     } catch (err) {
       console.error('WebLN payment failed:', err)
       throw err
+    } finally {
+      invoice.stopWaiting()
```

Comment:
Oh, maybe you could have just called `invoice.cancel` here instead of writing new code with `AbortController`?

Did you try that?